pyomo/contrib/piecewise/transform/incremental.py

```diff
@@ -0,0 +1,201 @@
+    triangulation must satisfy these properties:
+     (1) The simplices are ordered T_1, ..., T_N such that T_i has nonempty intersection
+         with T_{i+1}. It doesn't have to be a whole face; just a vertex is enough.
+     (2) On each simplex T_i, the vertices are ordered T_i^1, ..., T_i^n such
+         that T_i^n = T_{i+1}^1
+    """
+
+    CONFIG = PiecewiseLinearTransformationBase.CONFIG()
+    _transformation_name = "pw_linear_incremental"
+
+    # Implement to use PiecewiseLinearToGDP. This function returns the Var
+    # that replaces the transformed piecewise linear expr
+    def _transform_pw_linear_expr(self, pw_expr, pw_linear_func, transformation_block):
+        if pw_linear_func.triangulation not in (
+            Triangulation.OrderedJ1,
+            Triangulation.AssumeValid,
+        ):
+            # almost certain not to work
+            logging.getLogger('pyomo.contrib.piecewise.transform.incremental').error(
+                "Incremental transformation specified, but the triangulation may not be appropriately ordered. This is likely to lead to incorrect results!"
+            )
+        # Get a new Block() in transformation_block.transformed_functions, which
+        # is a Block(Any)
+        transBlock = transformation_block.transformed_functions[
+            len(transformation_block.transformed_functions)
+        ]
+
+        # Dimensionality of the PWLF
+        dimension = pw_expr.nargs()
+        transBlock.dimension_indices = RangeSet(0, dimension - 1)
+
+        # Substitute Var that will hold the value of the PWLE
+        substitute_var = transBlock.substitute_var = Var()
+        pw_linear_func.map_transformation_var(pw_expr, substitute_var)
+
+        # Bounds for the substitute_var that we will widen
+        substitute_var_lb = float("inf")
+        substitute_var_ub = -float("inf")
+
+        # Simplices are tuples of indices of points. Give them their own indices, too
+        simplices = pw_linear_func._simplices
+        num_simplices = len(simplices)
+        transBlock.simplex_indices = RangeSet(0, num_simplices - 1)
+        transBlock.simplex_indices_except_last = RangeSet(0, num_simplices - 2)
+        # Assumption: the simplices are really simplices and all have the same number of points,
+        # which is dimension + 1
+        transBlock.simplex_point_indices = RangeSet(0, dimension)
+        transBlock.nonzero_simplex_point_indices = RangeSet(1, dimension)
+        transBlock.last_simplex_point_index = Param(initialize=dimension)
+
+        # We don't seem to get a convenient opportunity later, so let's just widen
+        # the bounds here. All we need to do is go through the corners of each simplex.
+        for P, linear_func in zip(
+            transBlock.simplex_indices, pw_linear_func._linear_functions
+        ):
+            for v in transBlock.simplex_point_indices:
+                val = linear_func(*pw_linear_func._points[simplices[P][v]])
```

Comment:
I'm not sure. Most of these calls are only made twice, which is probably not worth it, but some are made n-ish times so it could be? My suspicion is that the immediately preceding call to get_ordered_j1_triangulation() will be much slower than anything in this file, but I might be wrong as I haven't benchmarked any of this.